web/source/osk/realizedGesture.interface.ts

```diff
@@ -0,0 +1,10 @@
+namespace com.keyman.osk {
+  export interface RealizedGesture {
+    readonly baseKey: KeyElement;
+    readonly promise: Promise<text.KeyEvent>;
```

Comment:
As gestures are events that occur over intervals of time, they are inherently asynchronous.  The simplest way to cleanly model this is with `Promise`s.

As a result, subkeys are now 'returned' to the OSK via `Promise`, which are then forwarded from within the resolution handler within `initGestures`.

This has the benefit of providing a very clear and clean definition of our expectations from any future implementation of additional gesture types.  Granted, there is a significant assumption to note regarding `multitap`, but we can cross that bridge when we come to it - and at least the expectation is made quite explicit.  Worst case... generic type parameters can help solve it.

Reply:
Can you add this comment (tweaked as required) as a code comment?

```plain
As gestures are events that occur over intervals of time, they are 
inherently asynchronous. The simplest way to cleanly model this 
is with Promises.

As a result, subkeys are 'returned' to the OSK via Promise, which 
are then forwarded from within the resolution handler within
initGestures.
```